Assets/Scripts/SS3D/Core/Systems/Rounds/RoundSystem.cs

```diff
@@ -0,0 +1,211 @@
+        }
+
+        private void HandleRequestStartRound(NetworkConnection conn, RequestStartRoundMessage m)
+        {
+            const ServerRoleTypes requiredRole = ServerRoleTypes.Administrator;
+
+            PlayerControlSystem playerControlSystem = GameSystems.PlayerControlSystem;
+            PermissionSystem permissionSystem = GameSystems.PermissionSystem;
+
+            string userCkey = playerControlSystem.GetSoulCkeyByConn(conn);
+            if (permissionSystem.GetUserPermission(userCkey) != requiredRole)
+            {
+                Debug.Log($"[{nameof(RoundSystem)}] - User {userCkey} doesn't have {requiredRole} permission");
+            }
+            else
+            {
+                ServerStartWarmup();   
+            }
+        }
+
+        /// <summary>
+        /// Server method to start the warmup
+        /// </summary>
+        [Server]
+        private void ServerStartWarmup()
+        {
+            if (!IsServer)
+            {
+                return;
+            }
+
+            // Starts the warmup
+            _currentTimerSeconds = _warmupTimerSeconds;
+            UpdateRoundState(RoundState.WarmingUp);
+            
+            _warmupCoroutine = StartCoroutine(TickWarmup());
+
+            WarmupStartedMessage warmupStartedMessage = new();
+            InstanceFinder.ServerManager.Broadcast(warmupStartedMessage);
+        }
+
+        [Server]
+        private void HandleStartRound()
+        {
+            // Only do SyncVar assignments, tick coroutine and the RPC on the server.
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (RoundRunning)
+            {
+                Debug.Log($"[{nameof(RoundSystem)}] - Can't start round as round is already running");
+                return;
+            }
+            
+            UpdateRoundState(RoundState.Starting);
+            
+            UpdateRoundState(RoundState.Running);
+            StopCoroutine(_warmupCoroutine);
+            _tickCoroutine = StartCoroutine(Tick());
+            
+            InstanceFinder.ServerManager.Broadcast(new RoundStartedMessage());
+        }
+
+        [Server]
+        private IEnumerator TickWarmup()
+        {
+            while (_currentTimerSeconds > 0)
+            {
+                UpdateClock(GetTimerSeconds());
+                Debug.Log($"[{nameof(RoundSystem)}] - Start timer: {_currentTimerSeconds}");
+                _currentTimerSeconds--;
+                yield return new WaitForSeconds(1);
+            }
+
+            HandleStartRound();
+        }
+
+        [Server]
+        private IEnumerator Tick()
+        {
+            if (!IsServer)
+            {
+                yield break;
+            }
+            
+            while (RoundRunning)
+            {
+                UpdateClock(GetTimerSeconds());
+                _currentTimerSeconds++;
```

Comment:
Depending on what you are using this timer for, this method may not be sufficiently accurate, as the time between these calls will always be slightly greater than one second. If the round lasts an hour and the server can handle 100 FPS, I would expect this method to lose about half a minute over the course of the round.